### PR TITLE
More precise `IntegerRangeType::toString()`

### PR DESCRIPTION
**Pull request by staabm**
requires https://github.com/phpstan/phpstan-src/pull/3474

### CODE REVIEW

File: tests/PHPStan/Analyser/nsrt/bug-7387.php [hunk 1]
```diff
@@ -56,15 +56,24 @@ public function positionalArgs($mixed, int $i, float $f, string $s, int $posInt,
...
+		assertType('non-falsy-string', sprintf("%2$'#14s", $mixed, $nonZeroIntRange));
 
 		assertType("non-falsy-string", sprintf('%2$14s', $mixed, 1));
 		assertType("non-falsy-string", sprintf('%2$14s', $mixed, '1'));
 		assertType("non-falsy-string", sprintf('%2$14s', $mixed, 'abc'));
+		assertType('numeric-string', sprintf('%2$14s', $mixed, $intRange));
 		assertType("'1'", sprintf('%2$s', $mixed, 1));
 		assertType("'1'", sprintf('%2$s', $mixed, '1'));
 		assertType("'abc'", sprintf('%2$s', $mixed, 'abc'));
+		assertType("'-1'|'0'|'1'|'2'|'3'|'4'|'5'", sprintf('%2$s', $mixed, $intRange));
```
Comment (staabm) [inline tests/PHPStan/Analyser/nsrt/bug-7387.php:82]: most important: we get now a constant type for the most common case

File: tests/PHPStan/Analyser/nsrt/bug-7387.php [hunk 2]
```diff
@@ -56,15 +56,24 @@ public function positionalArgs($mixed, int $i, float $f, string $s, int $posInt,
...
 		assertType('non-falsy-string&numeric-string', sprintf('%2$14s', $mixed, $negInt));
+		assertType('numeric-string', sprintf('%2$14s', $mixed, $nonZeroIntRange));
+
+		// https://3v4l.org/1ECIq
+		assertType('non-falsy-string', sprintf("%2$'#14s", $mixed, 1));
+		assertType('non-falsy-string', sprintf("%2$'#14s", $mixed, $i));
+		assertType('non-falsy-string', sprintf("%2$'#14s", $mixed, $posInt));
+		assertType('non-falsy-string', sprintf("%2$'#14s", $mixed, $negInt));
 		assertType('numeric-string', sprintf('%2$14s', $mixed, $intRange));
-		assertType('non-falsy-string&numeric-string', sprintf('%2$14s', $mixed, $nonZeroIntRange));
```
Comment (staabm) [inline tests/PHPStan/Analyser/nsrt/bug-7387.php:60]: we lost a bit precision in this edge-case

**Comment by ondrejmirtes** [comment]
Thank you!
